lib/extras/codec_apng.cc

```diff
@@ -187,6 +369,15 @@ int processing_finish(png_structp png_ptr, png_infop info_ptr) {
   }
 
   png_process_data(png_ptr, info_ptr, footer, 12);
+  // before destroying: check if we encountered any metadata chunks along the
+  // way
+  png_textp text_ptr;
+  int num_text;
+  png_get_text(png_ptr, info_ptr, &text_ptr, &num_text);
+  for (int i = 0; i < num_text; i++) {
+    (void)BlobsReaderPNG::Decode(text_ptr[i], metadata);
```

Comment:
Perhaps the return value could be used, it was used like this:

```
  if (!BlobsReaderPNG::Decode(state.s.info_png, &ppf->metadata)) {
    JXL_WARNING("PNG metadata may be incomplete");
  }
```

Reply:
Yes, but that function always returned `true` before, so that warning was never actually shown.

I changed the function to return `false` if it encounters some text chunk it doesn't understand, but that happens quite often it seems, so it results in a lot of warnings (e.g. png files often have simple text comments containing creation date, modify date, software used for editing, and things like that).